### PR TITLE
docs: Add documentation for RBD VolumeGroupSnapshot

### PR DESCRIPTION
**Pull request by black-dragon74**
This PR adds the sample YAMLs and documentation
for the RBD VolumeGroupSnapshot feature.

### CODE REVIEW

**Review by nixpanic — APPROVED** [review]
Looks good to me, thanks!

File: deploy/examples/csi/rbd/groupsnapshotclass.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+kind: VolumeGroupSnapshotClass
+metadata:
+  name: csi-rbdplugin-groupsnapclass
+driver: rook-ceph.rbd.csi.ceph.com # csi-provisioner-name
+parameters:
+  # Specify a string that identifies your cluster. Ceph CSI supports any
+  # unique string. When Ceph CSI is deployed by Rook use the Rook namespace,
+  # for example "rook-ceph".
+  clusterID: rook-ceph # namespace: cluster
+  pool: replicapool # eg. pool: <rbd-pool-name>
```
Comment (travisn) [inline deploy/examples/csi/rbd/groupsnapshotclass.yaml:12]:
```suggestion
  pool: replicapool
```

File: deploy/examples/csi/rbd/groupsnapshotclass.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
+apiVersion: groupsnapshot.storage.k8s.io/v1alpha1
```
[the remark this answers is not in the document]
Comment (black-dragon74) [reply deploy/examples/csi/rbd/groupsnapshotclass.yaml:2]: @nixpanic would know better :)

**Comment by Madhu-1** [comment]
@travisn the feature doesn't exists in cephcsi yet, its going to be part of the next release, we might need to remove the backport from 1.15

File: deploy/examples/csi/rbd/groupsnapshotclass.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Madhu-1) [reply deploy/examples/csi/rbd/groupsnapshotclass.yaml:2]: it will be in next 1.32 release timeline

**Comment by travisn** [comment]
> @travisn the feature doesn't exists in cephcsi yet, its going to be part of the next release, we might need to remove the backport from 1.15

Ah thanks for pointing out this important detail. Users frequently stumble onto the docs from master, so let's go ahead and revert from master and then backport the revert to 1.15. Then a new PR could be opened with these doc updates that we just put on hold until the next cephcsi is integrated with Rook. @black-dragon74 Could you open the revert PR? Thanks!

**Comment by black-dragon74** [comment]
> Ah thanks for pointing out this important detail. Users frequently stumble onto the docs from master, so let's go ahead and revert from master and then backport the revert to 1.15. Then a new PR could be opened with these doc updates that we just put on hold until the next cephcsi is integrated with Rook. @black-dragon74 Could you open the revert PR? Thanks!

Done at https://github.com/rook/rook/pull/14891 :)

**Pull request merged**